packages/material-ui/src/InputBase/InputBase.js

```diff
@@ -222,15 +222,22 @@ const InputBase = React.forwardRef(function InputBase(props, ref) {
     }
   };
 
+  React.useEffect(() => {
+    if (!isControlled) checkDirty(inputRef.current)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
What would rule add as deps?

Reply:
Looks like it would be checkDirty. I would prefer to add useCallback to `checkDirty` instead

Reply:
I added this as per the suggestions in https://github.com/mui-org/material-ui/pull/15446#discussion_r278873921

```diff
React.useEffect(() => {
    if (!isControlled) checkDirty(inputRef.current);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
}, []);
```

Because it wasn't allowing me to add `[]` & my understanding of React.useEffect is the if we add `[]` at the end. It will only call this method on the first time which is the equivalent of `componentDidMount`

Can you kindly tell as to the right approach here 😄  💭 🤔

Reply:
Removing the comment and adding checkDirty as a dependancy is probably the right thing to do. You then have to wrap checkDirty in useCallback and pass in a second argument. Eslint is helpful when working things out

Reply:
On converting `checkDirty` to `useCallback` I have to update one test

```diff
it('should fire the onEmpty muiFormControl and props callback when cleaned', () => {
        const handleEmpty = spy();
        wrapper.setProps({
          onEmpty: handleEmpty,
        });

        wrapper.find('input').instance().value = '';
        wrapper.find('input').simulate('change');
-       assert.strictEqual(handleEmpty.callCount, 1);
+       assert.strictEqual(handleEmpty.callCount, 2);
-       assert.strictEqual(muiFormControl.onEmpty.callCount, 1);
+       assert.strictEqual(muiFormControl.onEmpty.callCount, 3);
});
```

Is this the right behavior, should I commit this change? 😅

Reply:
Thank you @joshwooding for being super kind & helpful in helping so much for this ticket.  I understood some of it here https://github.com/mui-org/material-ui/pull/15446/commits/8feeba002f3bd0844deeb088eaae76ea702e8023 😅